core/src/main/java/org/apache/carbondata/core/datamap/DistributableDataMapFormat.java

```diff
@@ -85,33 +143,67 @@
       private ExtendedBlocklet currBlocklet;
       private List<DataMap> dataMaps;
 
-      @Override public void initialize(InputSplit inputSplit, TaskAttemptContext taskAttemptContext)
+      @Override
+      public void initialize(InputSplit inputSplit, TaskAttemptContext taskAttemptContext)
```

Comment:
Better create a new RecordReader for job cleaning purpose and override the initialize method for better separation.